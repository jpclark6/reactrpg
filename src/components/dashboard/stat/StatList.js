import React from 'react';

const StatList = (props) => {
    const { profile } = props;
    return profile.isEmpty ? <h4>Loading...</h4> :
     (
        <div className="brown-text text-darken-4">
            <h3 className="center" style={{margin: "20px 0 px 12px 0px"}}>{profile.firstName}'s Stats</h3>
            <h4 className="center" style={{margin: "12px"}}>Xp: {profile.attributes.xp}</h4>
            <h4 className="center" style={{margin: "12px"}}>Level: {profile.attributes.level}</h4>
            <table className="flow-text">
                <thead>
                    <tr>
                        <th>Skill</th>
                        <th>Points</th>
                    </tr>
                </thead>

                <tbody>
                    <tr>
                        <td>Strength</td>
                        <td>{profile.attributes.strength}</td>
                    </tr>
                    <tr>
                        <td>Intelligence</td>
                        <td>{profile.attributes.intelligence}</td>
                    </tr>
                    <tr>
                        <td>Stamina</td>
                        <td>{profile.attributes.stamina}</td>
                    </tr>
                    <tr>
                        <td>Grit</td>
                        <td>{profile.attributes.grit}</td>
                    </tr>
                    <tr>
                        <td>Karma</td>
                        <td>{profile.attributes.karma}</td>
                    </tr>
                    <tr>
                        <td>Luck</td>
                        <td>{profile.attributes.luck}</td>
                    </tr>
                </tbody>
            </table>
        </div>
    )
}

export default StatList;
