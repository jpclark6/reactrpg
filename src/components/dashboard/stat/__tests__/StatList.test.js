import React from 'react';
import StatList from '../StatList';
import renderer from 'react-test-renderer';

test('StatList is displayed', () => {
    const profile = {
        attributes: {
            grit: 0,
            intelligence: 0,
            karma: 0,
            level: 1,
            luck: 0,
            stamina: 0,
            strength: 0,
            xp: 0
        },
        firstName: "Bob",
        isEmpty: false,
        isLoaded: true,
        lastName: "Bobalo"
    }
    const tasks = [
        {
            id: '1',
            title: 'Example task',
            description: 'Load other tasks',
            reward: '',
            category: 'adulting',
            difficulty: 3,
            time: 1,
            recurring: 'none',
            completed: false,
            successful: false
        }
    ]
    const component = renderer.create(
        <StatList profile={profile} tasks={tasks} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});