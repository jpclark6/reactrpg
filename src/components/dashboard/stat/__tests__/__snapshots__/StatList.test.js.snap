// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatList is displayed 1`] = `
<div
  className="brown-text text-darken-4"
>
  <h3
    className="center"
    style={
      Object {
        "margin": "20px 0 px 12px 0px",
      }
    }
  >
    Bob
    's Stats
  </h3>
  <h4
    className="center"
    style={
      Object {
        "margin": "12px",
      }
    }
  >
    Xp: 
    0
  </h4>
  <h4
    className="center"
    style={
      Object {
        "margin": "12px",
      }
    }
  >
    Level: 
    1
  </h4>
  <table
    className="flow-text"
  >
    <thead>
      <tr>
        <th>
          Skill
        </th>
        <th>
          Points
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Strength
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td>
          Intelligence
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td>
          Stamina
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td>
          Grit
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td>
          Karma
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td>
          Luck
        </td>
        <td>
          0
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
